src/conditional.jl

```diff
@@ -195,11 +195,11 @@ function grad_apply_sparsity!(rbm::ConditionalRBM, X::Mat,
                   throw(ArgumentError("If :sparsity_cost is used, :sparsity_target should also be defined")))
     curr_sparsity = mean(hid_means(rbm, vis))
     penalty = cost * (curr_sparsity - target)
-    axpy!(-penalty, dW, dW)
-    axpy!(-penalty, dA, dA)
-    axpy!(-penalty, dB, dB)
-    axpy!(-penalty, db, db)
-    axpy!(-penalty, dc, dc)
+    axpy!(-penalty, ones(size(dW)), dW)
```

Comment:
Nice! I kind of expected as much cause `dW -= penalty` is still allocating another array rather than modifying the existing one. I'll update this PR with the changes.